src/BlueberryStaking.sol

```diff
@@ -655,33 +619,45 @@ contract BlueberryStaking is
 
         uint256 _totalRewardsAdded;
         uint256 _rewardDuration = rewardDuration;
-        
+
         uint256 _newTokensLength = _ibTokens.length;
         totalIbTokens += _newTokensLength;
 
         for (uint256 i; i < _newTokensLength; ++i) {
             address _ibToken = _ibTokens[i];
             uint256 _amount = _amounts[i];
 
+            if (_ibTokens[i] == address(0)) {
+                revert AddressZero();
+            }
+
             if (isIbToken[_ibToken]) {
-                revert BTokenAlreadyExists();
+                revert IBTokenAlreadyExists();
             }
 
             isIbToken[_ibToken] = true;
             ibTokens.push(_ibToken);
-            
+
             finishAt[_ibToken] = block.timestamp + _rewardDuration;
             _totalRewardsAdded += _amount;
 
             _setRewardRate(_ibToken, _amount, _rewardDuration);
 
-            emit IbTokenAdded(_ibToken, _amount, block.timestamp);
+            emit IbTokenAdded(_ibToken, _amount);
         }
 
         blb.transferFrom(msg.sender, address(this), _totalRewardsAdded);
     }
 
-    /// @inheritdoc IBlueberryStaking
+    /**
+     * @notice Called by the owner to change the reward rate for a given token(s)
+     * @dev uses address(0) in updateRewards as to not change the reward rate for any user but still update each
+     * mappings
+     * @dev the caller should consider the reward rate for each token before calling this function and total rewards
+     * should be less than the total amount of tokens
```

Comment:
What does this mean? All `_amounts` are net-new tokens entering the contract; the caller does not have to consider previous reward rates because leftovers are automatically propagated forward.

Reply:
Same as previous resolution. Old comments that are inaccurate. Will remove.